cluster/saltbase/salt/kube-addons/initd

```diff
@@ -0,0 +1,123 @@
+#!/bin/bash
+#
+### BEGIN INIT INFO
+# Provides:   kube-addons
+# Required-Start:    $local_fs $network $syslog kube-apiserver
+# Required-Stop:
+# Default-Start:     2 3 4 5
+# Default-Stop:      0 1 6
+# Short-Description: Kubernetes Addon Object Manager
+# Description:
+#   Enforces installation of Kubernetes Addon Objects
+### END INIT INFO
+
+
+# PATH should only include /usr/* if it runs after the mountnfs.sh script
+PATH=/sbin:/usr/sbin:/bin:/usr/bin
+DESC="Kubernetes Addon Object Manager"
+NAME=kube-addons
+DAEMON_LOG_FILE=/var/log/$NAME.log
+PIDFILE=/var/run/$NAME.pid
+SCRIPTNAME=/etc/init.d/$NAME
+KUBECTL=/usr/local/bin/kubectl
+
+# Define LSB log_* functions.
+# Depend on lsb-base (>= 3.2-14) to ensure that this file is present
+# and status_of_proc is working.
+. /lib/lsb/init-functions
+
+function addon_manager_async() {
+    # The business logic for whether a given object should be created
+    # was already enforced by salt, and /etc/kubernetes/addons is the
+    # managed result is of that. Start everything below that directory.
+    echo "== Kubernetes addon manager started at $(date -Is) =="
+    for obj in $(find /etc/kubernetes/addons -name \*.yaml); do
+	    ${KUBECTL} create -f ${obj} &
```

Comment:
The e2e tests catch DNS at the very least.

On Tue, Jan 20, 2015 at 10:41 AM, Joe Beda <notifications@github.com> wrote:

> In cluster/saltbase/salt/kube-addons/initd
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/3601#discussion-diff-23244133>
> :
>
> > +PIDFILE=/var/run/$NAME.pid
> > +SCRIPTNAME=/etc/init.d/$NAME
> > +KUBECTL=/usr/local/bin/kubectl
> > +
> > +# Define LSB log_* functions.
> > +# Depend on lsb-base (>= 3.2-14) to ensure that this file is present
> > +# and status_of_proc is working.
> > +. /lib/lsb/init-functions
> > +
> > +function addon_manager_async() {
> > +    # The business logic for whether a given object should be created
> > +    # was already enforced by salt, and /etc/kubernetes/addons is the
> > +    # managed result is of that. Start everything below that directory.
> > +    echo "== Kubernetes addon manager started at $(date -Is) =="
> > +    for obj in $(find /etc/kubernetes/addons -name \*.yaml); do
> > +	    ${KUBECTL} create -f ${obj} &
>
> NB: We are going to have to make sure that kubectl has the right auth
> material moving forward as we lock things down. Hopefully our e2e tests
> will catch this after this goes in.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/3601/files#r23244133>
> .
>